course/src/main/java/in/testpress/course/ui/ContentActivity.java

```diff
@@ -319,80 +328,112 @@ protected boolean shouldOverrideUrlLoading(Activity activity, String url) {
         };
         webView.setScrollBarStyle(WebView.SCROLLBARS_OUTSIDE_OVERLAY);
         fullScreenChromeClient = new FullScreenChromeClient(this);
-        chapterId = getIntent().getLongExtra(CHAPTER_ID, 0);
-        if (chapterId != 0) {
+        chapterId = getIntent().getLongExtra(CHAPTER_ID, -1);
+        if (chapterId != -1) {
             contents = getContentsFromDB();
         }
-        if (contents == null || contents.isEmpty()) {
-            contentId = getIntent().getStringExtra(CONTENT_ID);
-            if (contentId == null) {
-                Assert.assertNotNull("contentId must not be null.", contents);
-            } else {
+        contentId = getIntent().getLongExtra(CONTENT_ID, -1);
+
+        content = getContentFromDB();
+        if (contentId == -1) {
+            throw new IllegalArgumentException("contentId must not be null.");
+        }
+
+        if (chapterId == -1) {
+            if (content == null) {
                 updateContent();
+            } else {
+                checkContentType();
             }
             buttonLayout.setVisibility(View.GONE);
         } else {
-            position = getIntent().getIntExtra(POSITION, -1);
-            if (position == -1) {
-                throw new IllegalArgumentException("POSITION must not be null.");
+            if (content == null) {
+                updateContent();
+            } else {
+                contents = getContentsFromDB();
+                position = content.getOrder();
+                pageNumber.setText(String.format("%d/%d", position + 1, contents.size()));
+                checkContentType();
+                validateAdjacentNavigationButton();
             }
-            content = contents.get(position);
-            String title = getIntent().getStringExtra(ACTIONBAR_TITLE);
-            Assert.assertNotNull("ACTIONBAR_TITLE must not be null.", title);
-            getSupportActionBar().setTitle(title);
-            pageNumber.setText(String.format("%d/%d", position + 1, contents.size()));
-            checkContentType();
-            validateAdjacentNavigationButton();
+
+            exoplayerFullscreenHelper = new ExoplayerFullscreenHelper(this);
+            exoplayerFullscreenHelper.initializeOrientationListener();
+        }
+    }
+
+    private Content getContentFromDB() {
+        List<Content> contents = contentDao.queryBuilder()
+                .where(ContentDao.Properties.Id.eq(contentId)).list();
+        if (contents.isEmpty()) {
+            return null;
+        }
```

Comment:
This is inaccuracy. This activity should be deeplinkable. In that case it should load the content or show unauthorized.